src/components/CharacterWithQuote/CharacterWithQuote.js

```diff
@@ -0,0 +1,31 @@
+import React from 'react';
+import { withRouter } from 'react-router-dom';
+import PropTypes from 'prop-types';
+
+import { getQuote } from '../../utils/quotesService';
+import extractQuoteIDFromPath from '../../utils/extractQuoteIDFromPath';
+import extractActionFromPath from '../../utils/extractActionFromPath';
+import extractRoleFromPath from '../../utils/extractRoleFromPath';
+import { history } from '../../types';
+
+function CharacterWitQuote({ history, children }) {
```

Comment:
You can get `location` directly:

```
function Character({ location, children  }) {
  const {pathname} = location;
  ...
}
```

So _history_ type wont be needed as well.
Besides that QuoteBubble handles everything on its own, so we pretty much have to inject `pose` only.

Reply:
Yes, I removed history listener.
Fixed here: https://github.com/BuildIt-Poland/daily-routine/pull/75/commits/3a535146c57bfef269a3f85235bc413009812cad#diff-d1a8069f8bac699dcb367f774d57206a